app/src/components/common/GuestApplicationContext.tsx

```diff
@@ -0,0 +1,134 @@
+import {Box, Stack} from '@mui/material';
+import {Formik} from 'formik';
+import {useEffect, useState} from 'react';
+import {Outlet} from 'react-router-dom';
+import ProgressBar from './ProgressBar';
+import {GuestApplicationSchema} from '../../utils/GuestApplicationSchema';
+import {NavButtons} from './GuestApplicationButtons';
+import {useNavigate} from 'react-router-dom';
+
+export interface formInputValues {
+  fullName: string;
+  dateOfBirth: string;
+  gender: string;
+  email: string;
+  phoneNumber: string;
+  contactPrefrence: string;
+  nameOfGuest: Array<string>;
+  typeOfPet: Array<string>;
+  currentlyEmployed: string;
+  employmentSearch: string;
+  employmentDescription: string;
+  educationEnrollment: string;
+  educationEnrollmentSearch: string;
+  educationProgramName: string;
+  multilingual: string;
+  spokenLanguages: string;
+  cigarettesUse: string;
+  alcoholUse: string;
+  otherSubstanceUse: string;
+  mentalIllness: string;
+  programGoals: string;
+  hostRelationshipGoal: string;
+  potentialChallenges: string;
+}
+export const stepToRouteMapping: {[key: number]: string} = {
+  1: '',
+};
+export const initialValues = {
+  fullName: '',
+  dateOfBirth: '',
+  gender: '',
+  email: '',
+  phoneNumber: '',
+  contactPrefrence: '',
+  nameOfGuest: [''],
+  typeOfPet: [''],
+  currentlyEmployed: '',
+  employmentSearch: '',
+  employmentDescription: '',
+  educationEnrollment: '',
+  educationEnrollmentSearch: '',
+  educationProgramName: '',
+  multilingual: '',
+  spokenLanguages: '',
+  cigarettesUse: '',
+  alcoholUse: '',
+  otherSubstanceUse: '',
+  mentalIllness: '',
+  programGoals: '',
+  hostRelationshipGoal: '',
+  potentialChallenges: '',
+};
+
+export const GuestApplicationContext = () => {
+  const TOTALPAGES = 10; /*Once all forms are made and added to "stepToRouteMapping", make this number change based on "stepToRouteMapping"*/
+  const storedStep = localStorage.getItem('currentStep');
+  const initialStep = storedStep ? parseInt(storedStep, 10) : 1;
```

Comment:
Is the 10 here in reference to the TOTAL_PAGES variable?

Reply:
No it is not, ``praseInt`` takes a secondary argument to know which type of base value  to convert the string to (binary, decimal, octo). Just happens that 10 is for decimal. 10 is the default value, but I explicitly added it since there can be edge cases that it defaults to one of the other options.

Reply:
Ah thanks for clearing that up for me